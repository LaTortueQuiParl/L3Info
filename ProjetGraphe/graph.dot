graph {
	0 [label=0]
	1 [label=1]
	2 [label=2]
	3 [label=3]
	4 [label=4]
	5 [label=5]
	6 [label=6]
	7 [label=7]
	8 [label=8]
	9 [label=9]
	10 [label=10]
	11 [label=11]
	12 [label=12]
	13 [label=13]
	14 [label=14]
	15 [label=15]
	16 [label=16]
	17 [label=17]
	18 [label=18]
	19 [label=19]
	20 [label=20]
	21 [label=21]
	22 [label=22]
	23 [label=23]
	24 [label=24]
	25 [label=25]
	26 [label=26]
	27 [label=27]
	28 [label=28]
	29 [label=29]
	30 [label=30]
	31 [label=31]
	32 [label=32]
	33 [label=33]
	34 [label=34]
	35 [label=35]
	36 [label=36]
	37 [label=37]
	38 [label=38]
	39 [label=39]
	40 [label=40]
	41 [label=41]
	42 [label=42]
	43 [label=43]
	44 [label=44]
	45 [label=45]
	46 [label=46]
	47 [label=47]
	48 [label=48]
	49 [label=49]
	50 [label=50]
	51 [label=51]
	52 [label=52]
	53 [label=53]
	54 [label=54]
	55 [label=55]
	56 [label=56]
	57 [label=57]
	58 [label=58]
	59 [label=59]
	60 [label=60]
	61 [label=61]
	62 [label=62]
	63 [label=63]
	64 [label=64]
	65 [label=65]
	66 [label=66]
	67 [label=67]
	68 [label=68]
	69 [label=69]
	70 [label=70]
	71 [label=71]
	72 [label=72]
	73 [label=73]
	74 [label=74]
	75 [label=75]
	76 [label=76]
	77 [label=77]
	78 [label=78]
	79 [label=79]
	80 [label=80]
	81 [label=81]
	82 [label=82]
	83 [label=83]
	84 [label=84]
	85 [label=85]
	86 [label=86]
	87 [label=87]
	88 [label=88]
	89 [label=89]
	90 [label=90]
	91 [label=91]
	92 [label=92]
	93 [label=93]
	94 [label=94]
	95 [label=95]
	96 [label=96]
	97 [label=97]
	98 [label=98]
	37 -- 12
	72 -- 9
	75 -- 5
	79 -- 64
	16 -- 1
	76 -- 71
	6 -- 25
	50 -- 20
	18 -- 84
	11 -- 28
	29 -- 14
	50 -- 68
	87 -- 87
	94 -- 96
	86 -- 13
	9 -- 7
	63 -- 61
	22 -- 57
	1 -- 0
	60 -- 81
	8 -- 88
	13 -- 47
	72 -- 30
	71 -- 3
	70 -- 21
	49 -- 57
	3 -- 68
	24 -- 43
	76 -- 26
	52 -- 80
	41 -- 82
	15 -- 64
	68 -- 25
	98 -- 87
	7 -- 26
	25 -- 22
	9 -- 67
	23 -- 27
	37 -- 57
	83 -- 38
	8 -- 32
	34 -- 10
	23 -- 15
	87 -- 25
	71 -- 92
	74 -- 62
	46 -- 32
	88 -- 23
	55 -- 65
	77 -- 3
	0 -- 77
	6 -- 52
	85 -- 70
	2 -- 76
	91 -- 21
	75 -- 7
	77 -- 72
	75 -- 76
	43 -- 20
	30 -- 36
	7 -- 45
	68 -- 57
	82 -- 96
	13 -- 10
	23 -- 81
	7 -- 24
	74 -- 92
	20 -- 32
	12 -- 65
	94 -- 60
	24 -- 82
	97 -- 2
	92 -- 98
	10 -- 54
	96 -- 82
	86 -- 70
	66 -- 71
	48 -- 54
	15 -- 5
	17 -- 42
	20 -- 48
	22 -- 13
	97 -- 53
	84 -- 10
	96 -- 55
	61 -- 56
	89 -- 21
	96 -- 83
	25 -- 14
	13 -- 84
	43 -- 6
	77 -- 56
	59 -- 15
	24 -- 9
	66 -- 71
	53 -- 69
	36 -- 21
	40 -- 77
	91 -- 49
	47 -- 77
}
