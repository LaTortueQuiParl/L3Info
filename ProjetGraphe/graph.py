import networkx as nx
import matplotlib.pyplot as plt
import numpy as np


def genereGraphFromSnap():
    with open(f"graph.snap", "r") as graph_file:
        max = 0
        for line in graph_file.readlines():
            i,j = line.split(" ")
            j.strip("\n")
            if(int(i) > max):
                max = int(i)
            if(int(j) > max):
                max = int(j)

        graph = [i for i in range(max+1)]
        centres = [-1 for i in range(max+1)]
        G2.add_nodes_from(graph)
        G.add_nodes_from(graph)
        
        graph_file.seek(0)
        for line in graph_file.readlines():
            i,j = line.split(" ")
            j.strip("\n")
            G.add_edge(int(i), int(j))
            G2.add_edge(int(i), int(j))
    return centres, max

def dessinerGraphe():
    pos = nx.get_node_attributes(G,'pos')

    nx.draw(G, pos, with_labels=True, font_size=12,font_weight='bold')

    plt.axis("off")
    plt.show()
    #plt.savefig("path.png")

def degenerecence():
    degen = 1
    ok = False
    while(True):
        while(ok == False):
            index = []
            for i in G2.degree():
                if i[1] <= degen:
                    index.append(i[0])
                    centres[i[0]] = degen
            for i in index:
                G2.remove_node(i)

            if len(index) == 0:
                ok = True

        ok=False
        if len(G2.degree()) == 0:
            return degen
        else :
            degen+=1

def modifPos():

    for i in G.nodes():
        if centres[i] == max(centres):
            G.nodes[i]['pos'] = ((degenGraph-centres[i]+1)*np.cos((i*np.pi)/((centres.count(max(centres))+1)/2)), (degenGraph-centres[i]+1)*np.sin((i*np.pi)/((centres.count(max(centres))+1)/2)))
        else:
            G.nodes[i]['pos'] = ((degenGraph-centres[i]+1)*np.cos((i*np.pi)/((nbnoeuds+1)/2)), (degenGraph-centres[i]+1)*np.sin((i*np.pi)/((nbnoeuds+1)/2)))


G = nx.Graph()
G2 = nx.Graph()

centres, nbnoeuds = genereGraphFromSnap()
print(G)
degenGraph = degenerecence()
print("degenerecence = ", degenGraph)

print(G.nodes())
print(centres)

modifPos()

dessinerGraphe()


