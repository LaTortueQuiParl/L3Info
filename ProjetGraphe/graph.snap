37 12
72 9
75 5
79 64
16 1
76 71
6 25
50 20
18 84
11 28
29 14
50 68
87 87
94 96
86 13
9 7
63 61
22 57
1 0
60 81
8 88
13 47
72 30
71 3
70 21
49 57
3 68
24 43
76 26
52 80
41 82
15 64
68 25
98 87
7 26
25 22
9 67
23 27
37 57
83 38
8 32
34 10
23 15
87 25
71 92
74 62
46 32
88 23
55 65
77 3
0 77
6 52
85 70
2 76
91 21
75 7
77 72
75 76
43 20
30 36
7 45
68 57
82 96
13 10
23 81
7 24
74 92
20 32
12 65
94 60
24 82
97 2
92 98
10 54
96 82
86 70
66 71
48 54
15 5
17 42
20 48
22 13
97 53
84 10
96 55
61 56
89 21
96 83
25 14
13 84
43 6
77 56
59 15
24 9
66 71
53 69
36 21
40 77
91 49
47 77
